holonote/annotate/annotator.py

```diff
@@ -343,8 +343,50 @@ def get_display(self, *kdims: str | hv.Dimension) -> AnnotationDisplay:
             self._displays[element_key] = self._create_annotation_element(element_key)
         return self._displays[element_key]
 
-    def __mul__(self, other: hv.Element) -> hv.Overlay:
-        return other * self.get_element(*other.kdims)
+    def __mul__(self, other: hv.Element | hv.Layout | hv.Overlay | hv.NdOverlay) -> hv.Overlay:
+        if isinstance(other, (hv.Overlay, hv.NdOverlay)):
+            kdims, opts = other.kdims, other.opts.get().kwargs
+            if not kdims or kdims == ["Element"]:  # overlay and ndoverlay with no added kdims
+                # If no kdims in the overlay we use the first available
+                kdims = next(k for el in other.values() if (k := el.kdims))
+            return (
+                other * self.get_element(*[kdim for kdim in kdims if kdim.name in self.spec])
+            ).opts(**opts)
+        elif isinstance(other, hv.Layout):
+            opts = other.opts.get().kwargs
+            to_layout = []
+            for el in other:
+                kdims = el.kdims
+                el_opts = el.opts.get().kwargs
+                if not el.kdims or el.kdims == ["Element"]:
+                    try:
+                        kdims = next(k for sub_el in el.values() if (k := sub_el.kdims))
+                    except (
+                        StopIteration
+                    ):  # if the dmap hasn't been rendered yet, render to access kdims
+                        hv.render(el)
```

Comment:
This is probably not the right way to do it, but I couldn't figure out how else to get access to kdims from a dynamicmap, other than to render it